.github/workflows/container-publish.yml

```diff
@@ -24,34 +27,49 @@ jobs:
     - name: Figure out the tag based on event trigger
       env:
         TRIGGER: ${{ github.event_name }}
-      run: | 
+      run: |
         if [ "$TRIGGER" == "workflow_dispatch" ]; then
           echo "TRIGGERED_BY_WORKFLOW_DISPATCH with build tag ${{ github.event.inputs.buildTag }}"
           BUILD_TAG=${{ github.event.inputs.buildTag }}
         else
-          BUILD_TAG=${{ steps.meta.outputs.tags }}
+          # on push to a tag, use the latest tag starting with v and latest as the build tag
+          BUILD_TAG="$(git tag --list --sort=version:refname 'v*' | tail -1),latest"
+        fi
+
+        CONTAINER_TAGS=${{ github.event.inputs.containerTags }}
+        if [ -z $CONTAINER_TAGS ]; then
+          CONTAINER_TAGS=$BUILD_TAG
         fi
+
+        # Join all tags with the image name
+        TAGS=""
+        for TAG in $(echo $CONTAINER_TAGS|tr ',' ' ') ; do \
+          TAGS=${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}:${TAG},${TAGS}
+        done
+
+        # Remove trailing comma
+        TAGS=$(echo $TAGS | sed s/,$//g)
+
         echo "BUILD_TAG=$BUILD_TAG" >> $GITHUB_ENV
-        echo "using build tag ${{ env.BUILD_TAG }}"
+        echo "CONTAINER_TAGS=$TAGS" >> $GITHUB_ENV
+    - name: Print finalised tags
```

Comment:
created a new step to print the tags because the values written to `$GITHUB_ENV` are available only in the next steps.